service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -0,0 +1,175 @@
+package no.nav.su.se.bakover.service.revurdering
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.getOrHandle
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.client.person.MicrosoftGraphApiOppslag
+import no.nav.su.se.bakover.common.log
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.database.RevurderingRepo
+import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.behandling.Behandling
+import no.nav.su.se.bakover.domain.behandling.BeregnetRevurdering
+import no.nav.su.se.bakover.domain.behandling.OpprettetRevurdering
+import no.nav.su.se.bakover.domain.behandling.Revurdering
+import no.nav.su.se.bakover.domain.behandling.RevurderingTilAttestering
+import no.nav.su.se.bakover.domain.behandling.SimulertRevurdering
+import no.nav.su.se.bakover.domain.beregning.Beregning
+import no.nav.su.se.bakover.domain.beregning.fradrag.Fradrag
+import no.nav.su.se.bakover.domain.brev.LagBrevRequest
+import no.nav.su.se.bakover.domain.oppgave.OppgaveConfig
+import no.nav.su.se.bakover.service.brev.BrevService
+import no.nav.su.se.bakover.service.oppgave.OppgaveService
+import no.nav.su.se.bakover.service.person.PersonService
+import no.nav.su.se.bakover.service.sak.SakService
+import no.nav.su.se.bakover.service.utbetaling.UtbetalingService
+import java.util.UUID
+
+internal class RevurderingServiceImpl(
+    private val sakService: SakService,
+    private val utbetalingService: UtbetalingService,
+    private val revurderingRepo: RevurderingRepo,
+    private val oppgaveService: OppgaveService,
+    private val personService: PersonService,
+    private val microsoftGraphApiClient: MicrosoftGraphApiOppslag,
+    private val brevService: BrevService,
+) : RevurderingService {
+
+    override fun opprettRevurdering(
+        sakId: UUID,
+        periode: Periode,
+        saksbehandler: NavIdentBruker.Saksbehandler
+    ): Either<KunneIkkeRevurdere, Revurdering> {
+        if (!periode.erPeriodenIMånederEtter()) return KunneIkkeRevurdere.KanIkkeRevurdereInneværendeMånedEllerTidligere.left()
+
+        return hentSak(sakId)
+            .map { sak ->
+                val tilRevurdering = sak.behandlinger()
+                    .filter { it.status() == Behandling.BehandlingsStatus.IVERKSATT_INNVILGET }
+                    .filter { it.beregning()!!.getPeriode() inneholder periode }
+
+                if (tilRevurdering.isEmpty()) return KunneIkkeRevurdere.FantIngentingSomKanRevurderes.left()
+                if (tilRevurdering.size > 1) return KunneIkkeRevurdere.KanIkkeRevurderePerioderMedFlereAktiveStønadsperioder.left()
```

Comment:
Tenker vi gjøre noen sjekk på om det finnes andre aktive revurderinger for denne behandlingen?

Reply:
Tenker det letteste og starte med er bare å ha 1 aktiv.